### ISSUE TITLE
Objecto vs Clase

### ISSUE THREAD

**Issue by PalumboN** [ayalae/DDS-TP1-Macowins#2]
Todos los estados de las prendas están definidos como `object`s:
```wollok
object nuevo {
    method precio(preciobase) = preciobase
}

object promocion {
    var property valorFijo
    method precio(preciobase) = preciobase - valorFijo
}

object liquidacion {
    method precio(preciobase) = preciobase/2
}
```

Esto significa que _existirá solamente un objeto de cada `estado` en todo el sistema_, lo que implica que estos objetos **serán compartidos entre las prendas**. Ejemplo:
![prendas](https://user-images.githubusercontent.com/4098184/78027083-23d66180-7333-11ea-8769-b8bdb7de9ed4.png)

- Para los casos de prendas _nuevas_ o en _liquidación_  esto es indiferente, porque dichos objetos **no tienen estado** (_stateless_) o sea atributos.
- Pero en los casos de prendas con _promoción_, esta decisión implica que **todas ellas tendrán el mismo descuento**, ya que todas apuntan al mismo objeto `promocion` que tiene el `valorFijo` (o descuento)

Consecuencia de esto, este diseño puede llegar a ser **bastante restrictivo**, ya que _solamente soporta tener un descuento de promoción para todas las prendas_. Esta es una pregunta que estaría bueno preguntársela al cliente si tenemos dudas, aunque, por contexto, probablemente queramos poder configurar un descuento para cada (conjunto de) prenda.

**Comment by ayalae** [comment]
Claro, lo comprendo, claramente el objeto `promocion` deberia ser una clase para poder tener distintos descuentos por prenda. Muchas gracias por la corrección!
